drivers/wireless/ieee80211/bcm43xxx/bcmf_sdio.h

```diff
@@ -110,13 +111,22 @@ struct bcmf_sdio_dev_s
 
 /* Structure used to manage SDIO frames */
 
+struct __attribute__((packed)) bcmf_dma_buf
+{
+  uint8_t align_padding[CONFIG_IEEE80211_BROADCOM_SDIO_DMA_BUF_ALIGNMENT -
+                        FIRST_WORD_SIZE];
+  uint8_t first_word[FIRST_WORD_SIZE];
+  uint8_t remaining_aligned_buf[HEADER_SIZE - FIRST_WORD_SIZE +
+                                MAX_NETDEV_PKTSIZE + CONFIG_NET_GUARDSIZE];
+};
+
 struct bcmf_sdio_frame
 {
   struct bcmf_frame_s header;
   bool                tx;
   dq_entry_t          list_entry;
-  uint8_t             data[HEADER_SIZE + MAX_NETDEV_PKTSIZE +
-                           CONFIG_NET_GUARDSIZE];
+  struct bcmf_dma_buf data
+  __attribute__((aligned(CONFIG_IEEE80211_BROADCOM_SDIO_DMA_BUF_ALIGNMENT)));
```

Comment:
directly use aligned_data:
https://github.com/apache/incubator-nuttx/pull/4238

Reply:
Good. I will use the new aligned_data as soon as #4238 merged into the master branch.

Reply:
GCC attributes are forbidden in common code.  This should not be merged until this fixed.  There are compiler-independent macros in include/nuttx/compiler.h that should be used to make the alignment compiler independent.

Reply:
Hi @patacongo, good point. Thanks.

@xiaoxiang781216, initially I've misunderstood, I thought you've introduced the aligned_data macro just today in #4238.
As the aligned_data macro is already in the master branch, now I've replaced __attribute__((aligned(x))) by aligned_data(x).

Concerning compiler independence, now I've found "__attribute__((packed))" occurrences in bcm43xxx driver.
Thus I've replaced "__attribute__((packed))" by "begin_packed_struct / end_packed_struct" everywhere in bcm43xxx driver as well.

Reply:
#4238 will fix all ```__attribute__``` in a batch, so you can drop the related change from your PR.